File: examples/programming/p2.js
"use strict"

var botnana = require('../../index')

function test() {
    // Show reponse data
    botnana.debug_level = 1;
    botnana.once("ready", function() {
        // Programming API                
        var Program = botnana.Program;
        // Program p2
        var p2 = new Program("p2");
        var s1 = p2.ethercat.slave(1);
        s1.hm();
        s1.go();
        s1.pp();
        s1.move_to(30000);
        s1.go();
        p2.deploy();
        // Run Program p1
        botnana.once("deployed", function() { p2.run(); });
    })
    botnana.start("ws://192.168.7.2:3012");
}

test();

File: botnanars/examples/p2.rs
extern crate botnanars;
use botnanars::botnana;
use std::sync::{Arc, Mutex};

fn main() {
    let botnana = Arc::new(Mutex::new(botnana::botnana::new().unwrap()));

    let btn = botnana.clone();
    botnana.lock().unwrap().once("ready", move |_| {
        let mut p = btn.lock().unwrap().program("p2");
        let s1 = p.ethercat.slave(1);
        match s1 {
            Some(mut s) => {
                s.hm();
                s.go();
                s.pp();
                s.move_to(30000);
                s.go();
            }
            None => {}
        }

        p.deploy();
    });

    let btn = botnana.clone();
    botnana.lock().unwrap().once("deployed", move |_| {
        let p = btn.lock().unwrap().program("p2");
        p.run();
    });

    botnana.lock().unwrap().start("ws://192.168.50.197:3012");
    loop {}
}
